xapian-letor/ranker/listmle_ranker.cc

```diff
@@ -140,7 +140,9 @@ ListMLERanker::train(const vector<FeatureVector>& training_data)
     }
 
     for (int iter_num = 1; iter_num <= iterations; ++iter_num) {
-	batch_learning(training_data, new_parameters, learning_rate);
+	for (size_t it = 0; it < training_data.size(); ++it) {
+	    batch_learning(training_data[it], new_parameters, learning_rate);
+	}
```

Comment:
You can use C++11 range `for` here, which is more readable and eliminates the explicit index:

```
    for (auto& item : training_data) {
        batch_learning(item, new_parameters, learning_rate);
    }
```

Reply:
fixed in a fixup commit.